### PR TITLE
build: bump k256 to 0.13.2

### LINKED ISSUE #70: Upgrade `k256` in _rust-k256_ to "~0.13.0"

### PR DESCRIPTION
**Pull request by RajeshRk18**
Resolves #70

### CODE REVIEW

File: rust-k256/src/lib.rs [hunk 1]
```diff
@@ -106,7 +105,7 @@ impl PlumeSignature<'_> {
         // don't forget to check `c` is `Output<Sha256>` in the #API
         let c = Output::<Sha256>::from_slice(self.c);
         // TODO should we allow `c` input greater than BaseField::MODULUS?
-        let c_scalar = &Scalar::from_uint_reduced(U256::from_be_byte_array(c.to_owned()));
```
Comment (skaunov) [inline rust-k256/src/lib.rs:109]: I wonder how `clone` is better? Impact is diminishing ofc, but just for the sake of discussion.

Comment (skaunov) [reply rust-k256/src/lib.rs:109]: (I have changed to `to_owned` to ensure that `c` is "eaten" at that point and can't be used further.)

Comment (RajeshRk18) [reply rust-k256/src/lib.rs:109]: Yep. Didn't check if that's used further or not.
Will change it!

Comment (RajeshRk18) [reply rust-k256/src/lib.rs:109]: Hey just now checked that `c` is indeed used later so clone makes sense. But the value to be cloned itself is borrowed one. In that sense, to_owned makes more sense.

Comment (skaunov) [reply rust-k256/src/lib.rs:109]: I guess it's so, was commenting from the top of my head! %)

File: rust-k256/src/lib.rs [hunk 2]
```diff
@@ -106,7 +105,7 @@ impl PlumeSignature<'_> {
...
+        let c_scalar = &Scalar::reduce_nonzero(U256::from_be_byte_array(c.to_owned()));
```
Comment (0xmad) [inline rust-k256/src/lib.rs:108]: I think you need to run `cargo fmt` and `cargo clippy`.

**Comment by 0xmad** [comment]
@RajeshRk18 please rebase with current main branch so PR has only commits related to this issues.

File: rust-k256/src/lib.rs [hunk 3]
```diff
@@ -106,24 +105,20 @@ impl PlumeSignature<'_> {
...
```
Comment (skaunov) [inline rust-k256/src/lib.rs:108]: Looks like `reduce_nonzero_bytes` will be more appropriate here, does it?

Also IIRC @Divide-By-0 mentioned that `c_scalar` indeed shouldn't be zero, and the update renders next check excessive (remove it, pls).

**Review by skaunov — DISMISSED** [review]

File: rust-k256/src/lib.rs [hunk 4]
```diff
@@ -106,24 +105,20 @@ impl PlumeSignature<'_> {
...
+        let hashed_to_curve_r = hashed_to_curve * self.s - self.nullifier * c_scalar;
 
         // Check if the given hash matches
         let result = |components: Vec<&ProjectivePoint>| -> bool {
-            if &c_sha256_vec_signal(components) == c {
-                true
-            } else {
-                false
-            }
+            &c_sha256_vec_signal(components) == c
```
[the remark this answers is not in the document]
Comment (RajeshRk18) [reply rust-k256/src/lib.rs:121]: just to make it simple yes

[the remark this answers is not in the document]
Comment (skaunov) [reply rust-k256/src/lib.rs:121]: (Just to be on the same page: I meant ditching the whole `result` closure.)

[the remark this answers is not in the document]
Comment (RajeshRk18) [reply rust-k256/src/lib.rs:121]: I think it's fine

File: rust-k256/src/lib.rs (hunk 3 above)
Comment (skaunov) [reply rust-k256/src/lib.rs:108]: I've looked into this one a bit deeper as @Divide-By-0 asked to make PRs merge easier, and I realize my comment isn't quite correct. I perceived `reduce_nonzero` as returning a non-zero `Scalar`, but actually it requires non-zero input. I believe the best option here is to use `NonZeroScalar`; or at least to check the value before the call.

**Pull request merged**